src/backend/protocol.ts

```diff
@@ -2,91 +2,177 @@ import { dialog } from 'electron'
 import { logError, logInfo, LogPrefix } from './logger/logger'
 import i18next from 'i18next'
 import { getInfo } from './utils'
-import { Runner } from 'common/types'
+import { GameInfo, Runner, SideloadGame } from 'common/types'
 import { getMainWindow, sendFrontendMessage } from './main_window'
+import { icon } from './constants'
 
+type Command = 'ping' | 'launch'
+
+const RUNNERS = ['legendary', 'gog', 'sideload']
+
+/**
+ * Handles a protocol request
+ * @param args The args to search
+ * @example
+ * handleProtocol(['heroic://ping'])
+ * // => 'Received ping! Arg: undefined'
+ * handleProtocol(['heroic://launch/gog/123'])
+ * // => 'Received launch! Runner: gog, Arg: 123'
+ **/
 export async function handleProtocol(args: string[]) {
   const mainWindow = getMainWindow()
 
-  // Figure out which argv element is our protocol
-  let url = ''
-  args.forEach((val) => {
-    if (val.startsWith('heroic://')) {
-      url = val
-    }
-  })
-
-  const [scheme, path] = url.split('://')
-  if (!url || scheme !== 'heroic' || !path) {
+  const url = getUrl(args)
+  if (!url) {
     return
   }
-  let [command, arg] = path.split('/')
-  if (!command || !arg) {
-    command = path
-    arg = ''
-  }
+
+  const [command, runner, arg = ''] = parseUrl(url)
 
   logInfo(`received '${url}'`, LogPrefix.ProtocolHandler)
 
-  if (command === 'ping') {
-    return logInfo(['Received ping! Arg:', arg], LogPrefix.ProtocolHandler)
+  switch (command) {
+    case 'ping':
+      return handlePing(arg)
+    case 'launch':
+      await handleLaunch(runner, arg, mainWindow)
+      break
+    default:
+      return
   }
+}
 
-  if (command === 'launch') {
-    const runners: Runner[] = ['legendary', 'gog', 'sideload']
+/**
+ * Gets the url from the args
+ * @param args The args to search
+ * @returns The url if found, undefined otherwise
+ * @example
+ * getUrl(['heroic://ping'])
+ * // => 'heroic://ping'
+ * getUrl(['heroic://launch/gog/123'])
+ * // => 'heroic://launch/gog/123'
+ * getUrl(['heroic://launch/legendary/123'])
+ * // => 'heroic://launch/legendary/123'
+ **/
+function getUrl(args: string[]): string | undefined {
+  return args.find((arg) => arg.startsWith('heroic://'))
+}
+
+/**
+ * Parses a url into a tuple of [Command, Runner?, string?]
+ * @param url The url to parse
+ * @returns A tuple of [Command, Runner?, string?]
+ * @example
+ * parseUrl('heroic://ping')
+ * // => ['ping', undefined, undefined]
+ * parseUrl('heroic://launch/gog/123')
+ * // => ['launch', 'gog', '123']
+ * parseUrl('heroic://launch/legendary/123')
+ * // => ['launch', 'legendary', '123']
```

Comment:
maybe add an example of the old launch command? I understand it would be `heroic://launch/123` => `['launch', undefined, '123']`

Reply:
done